src/Neo.VM/ReferenceCounter/ReferenceCounterV2.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ReferenceCounterV2.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.VM.Types;
+using System;
+
+namespace Neo.VM
+{
+    /// <summary>
+    /// Used for reference counting of objects in the VM.
+    /// </summary>
+    public sealed class ReferenceCounterV2 : IReferenceCounter
+    {
+        public RCVersion Version { get; } = RCVersion.V2;
+
+        private readonly ExecutionEngineLimits _limits;
+
+        // Keeps the total count of references.
+        private int _referencesCount = 0;
```

Comment:
Regarding this: https://github.com/neo-project/neo/pull/3581#discussion_r1870242900

What do you think about making it unsigned integer, as @roman-khimov suggested and as we have it in NeoGo? Then there won't be any compatibility issues.